flink-table/flink-table-common/src/main/java/org/apache/flink/table/expressions/ValueLiteralExpression.java

```diff
@@ -219,6 +222,83 @@ public String asSummaryString() {
         return stringifyValue(value);
     }
 
+    @Override
+    public String asSerializableString() {
+        if (value == null && !dataType.getLogicalType().is(LogicalTypeRoot.NULL)) {
+            return String.format(
+                    "CAST(NULL AS %s)",
+                    // casting does not support nullability
+                    dataType.getLogicalType().copy(true).asSerializableString());
+        }
+        final LogicalType logicalType = dataType.getLogicalType();
+        switch (logicalType.getTypeRoot()) {
+            case TINYINT:
+                return String.format("CAST(%s AS TINYINT)", value);
+            case SMALLINT:
+                return String.format("CAST(%s AS SMALLINT)", value);
+            case BIGINT:
+                return String.format("CAST(%s AS BIGINT)", value);
+            case FLOAT:
+                return String.format("CAST(%s AS FLOAT)", value);
+            case DOUBLE:
+                return String.format("CAST(%s AS DOUBLE)", value);
+            case CHAR:
+            case VARCHAR:
+            case DECIMAL:
+            case INTEGER:
+                return stringifyValue(value);
+            case BOOLEAN:
+            case SYMBOL:
+            case NULL:
+                return stringifyValue(value).toUpperCase(Locale.ROOT);
+            case BINARY:
+            case VARBINARY:
+                return String.format("X'%s'", StringUtils.byteToHexString((byte[]) value));
+            case DATE:
+                return String.format("DATE '%s'", getValueAs(LocalDate.class).get());
+            case TIME_WITHOUT_TIME_ZONE:
+                return String.format("TIME '%s'", getValueAs(LocalTime.class).get());
+            case TIMESTAMP_WITHOUT_TIME_ZONE:
+                final LocalDateTime localDateTime = getValueAs(LocalDateTime.class).get();
+                return String.format(
+                        "TIMESTAMP '%s %s'",
+                        localDateTime.toLocalDate(), localDateTime.toLocalTime());
+            case TIMESTAMP_WITH_LOCAL_TIME_ZONE:
+                final Instant instant = getValueAs(Instant.class).get();
+                return String.format("TO_TIMESTAMP_LTZ(%d, %d)", instant.toEpochMilli(), 3);
```

Comment:
we can support more than milliseconds here

Reply:
Actually we support only precision of (3): https://github.com/apache/flink/blob/53ece12c25579497338ed59a7aebe70f2b3d9ed6/flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/functions/sql/FlinkSqlOperatorTable.java#L816

https://github.com/apache/flink/blob/84044f45830542d13d6c9baeb2bfe30de3eac4ac/flink-table/flink-table-common/src/main/java/org/apache/flink/table/utils/DateTimeUtils.java#L328

Reply:
I added a check to throw an exception if provided a greater precision.